clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -524,7 +524,7 @@ static KafkaAdminClient createInternal(AdminClientConfig config, TimeoutProcesso
                 config.getLong(AdminClientConfig.RECONNECT_BACKOFF_MAX_MS_CONFIG),
                 config.getInt(AdminClientConfig.SEND_BUFFER_CONFIG),
                 config.getInt(AdminClientConfig.RECEIVE_BUFFER_CONFIG),
-                (int) TimeUnit.HOURS.toMillis(1),
+                config.getInt(AdminClientConfig.REQUEST_TIMEOUT_MS_CONFIG),
```

Comment:
Should this be a long like the other timeouts?

Reply:
It seems we don't make it very clear about when to use Int and when to use Long, but `request.timeout.ms` is defined as Integer, the code can be found in :
```
.define(REQUEST_TIMEOUT_MS_CONFIG,
    Type.INT,
    30000,
    atLeast(0),
    Importance.MEDIUM,
    REQUEST_TIMEOUT_MS_DOC)
```